apps/network-alerts/js/attention/notify.js

```diff
@@ -0,0 +1,116 @@
+/* global
+  AudioContext
+ */
+
+(function(exports) {
+  'use strict';
+
+  const SETTINGS = {
+    notificationVolume: 'audio.volume.notification',
+    vibration: 'vibration.enabled'
+  };
+
+  const ATTENTION_PATTERN = [2000, 500, 1000, 500, 1000, 500];
+  ATTENTION_PATTERN.push(...ATTENTION_PATTERN);
+
+  const ATTENTION_SOUND_VOLUME = 0.3;
+
+  function getSetting(key) {
+    if (!navigator.mozSettings) {
+      return Promise.reject(new Error('The mozSettings API is not available.'));
+    }
+
+    return navigator.mozSettings.createLock().get(key).then(
+      result => result[key]
```

Comment:
nit: haven't we agreed to always keep round brackets around arrow function parameters? IIRC you liked it more, but I'm fine with both approaches :)